### PR TITLE
add org.deluge_torrent.deluge

### PR DESCRIPTION
**Pull request by fastrizwaan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages. **YES**
- [x] My pull request follows the instructions at [App Submission][submission]. **YES**
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* **YES**
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type. **YES**
- [x] I am an upstream contributor to the project. If not, **I contacted upstream developers** about submitting their software to Flathub. **Link:https://dev.deluge-torrent.org/ticket/3444**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). **Application developer own the domain**
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.deluge_torrent.deluge

**Comment by fastrizwaan** [comment]
bot, build org.deluge_torrent.deluge

**Comment by fastrizwaan** [comment]
bot, build org.deluge_torrent.deluge

**Comment by fastrizwaan** [comment]
any update?

File: main_window.ui [hunk 1]
```diff
@@ -0,0 +1,796 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline main_window.ui:1]: Is the file needed?

File: org.deluge_torrent.deluge.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.metainfo.xml:1]: should be contributed upstream if possible.

File: org.deluge_torrent.deluge.yaml [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.PowerManagement
+  - --talk-name=org.gnome.SessionManager
+  - --own-name=org.kde.StatusNotifierItem-2-2
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.yaml:26]: This won't work

File: org.deluge_torrent.deluge.yaml [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+modules:
+
+- name: boost
+  buildsystem: simple
+  sources:
+  - type: archive
+    url: https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.bz2
+    sha256: 953db31e016db7bb207f11432bef7df100516eeb746843fa0486a222e3fd49cb
+  build-commands:
+  - ./bootstrap.sh --prefix="${FLATPAK_DEST}" --with-python=python3
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.yaml:40]: You should build only what you need from boost.

File: org.deluge_torrent.deluge.yaml [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+  build-commands:
+  - python3 setup.py build
+  - python3 setup.py egg_info
+  - python3 setup.py install --prefix=/app --root=/
+  sources:
+  - type: archive
+    url: https://files.pythonhosted.org/packages/b9/2e/64db92e53b86efccfaea71321f597fa2e1b2bd3853d8ce658568f7a13094/MarkupSafe-1.1.1.tar.gz
+    sha256: 29872e92839765e546828bb7754a68c418d927cd064fd4708fab9fe9c8bb116b
+
+- name: mako
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.yaml:407]: You can generate most of the python dependencies using https://github.com/flatpak/flatpak-builder-tools/tree/master/pip

File: org.deluge_torrent.deluge.yaml [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+  build-commands:
+  - python3 setup.py build
+  - python3 setup.py egg_info
+  - python3 setup.py install --prefix=/app --root=/
+  post-install:
+  - install -Dm644 /app/share/icons/hicolor/128x128/apps/deluge.png /app/share/icons/hicolor/128x128/apps/org.deluge_torrent.deluge.png
+  - mkdir -p /app/share/applications
+  - install -Dm644 deluge/ui/data/share/applications/deluge.desktop.in /app/share/applications/org.deluge_torrent.deluge.desktop
+  - sed 's/^_//g' -i /app/share/applications/org.deluge_torrent.deluge.desktop
+  - sed 's/Icon=deluge/Icon=org.deluge_torrent.deluge/g' -i /app/share/applications/org.deluge_torrent.deluge.desktop
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.yaml:440]: Use desktop-file-edit?

File: org.deluge_torrent.deluge.yaml [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+  
+  - install -Dm644 main_window.ui /app/lib/python3.8/site-packages/deluge/ui/gtk3/glade/main_window.ui
```
Comment (bilelmoussaoui) [inline org.deluge_torrent.deluge.yaml:442]: Not sure why you need this?

File: README.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+
+# deluge_flatpak
```
Comment (bilelmoussaoui) [inline README.md:2]: Most of this is a duplicate of the docs. Can be removed

**Review by fastrizwaan — COMMENTED** [review]
Thank you for guiding me!

**Comment by fastrizwaan** [comment]
bot, build org.deluge_torrent.deluge

**Review by fastrizwaan — COMMENTED** [review]
Please review. Thanks.

File: README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+Deluge is a lightweight, Free Software, cross-platform BitTorrent client. See: https://deluge-torrent.org/
```
Comment (barthalion) [inline README.md:1]: No need to have it here.

File: org.deluge_torrent.deluge.yaml [hunk 10]
```diff
@@ -0,0 +1,437 @@
...
+  config-opts:
+    - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    - -Dpython-bindings=ON 
+    - -Dboost-python-module-name=python
+  sources:
+    - type: archive
+      url: https://github.com/arvidn/libtorrent/releases/download/v1.2.12/libtorrent-rasterbar-1.2.12.tar.gz
+      sha256: c3744ac9fa41f6e6ebf79538a2ea678df76a2cbbaf3ac6ae2c05455314e5cce8
+
+## Python3 modules
```
Comment (barthalion) [inline org.deluge_torrent.deluge.yaml:42]: FYI you can use https://github.com/flatpak/flatpak-builder-tools/tree/master/pip to generate python dependencies.

File: org.deluge_torrent.deluge.yaml [hunk 11]
```diff
@@ -0,0 +1,437 @@
...
+- name: deluge
+  buildsystem: simple
+  build-commands:
+  - python3 setup.py build
+  - python3 setup.py egg_info
+  - python3 setup.py install --prefix=/app --root=/
+  post-install:
+  - install -Dm644 /app/share/icons/hicolor/128x128/apps/deluge.png /app/share/icons/hicolor/128x128/apps/org.deluge_torrent.deluge.png
+  - mkdir -p /app/share/applications
+  - install -Dm644 deluge/ui/data/share/applications/deluge.desktop.in /app/share/applications/org.deluge_torrent.deluge.desktop
```
Comment (barthalion) [inline org.deluge_torrent.deluge.yaml:424]: I think I prefer rename-icon and copy-icon over this.

File: org.deluge_torrent.deluge.yaml [hunk 12]
```diff
@@ -0,0 +1,437 @@
...
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications
+  - --filesystem=xdg-download
+
+modules:
```
Comment (barthalion) [inline org.deluge_torrent.deluge.yaml:17]: You should add a global or per-module `cleanup` array to remove development headers and static libraries.

File: org.deluge_torrent.deluge.yaml (hunk 10 above)
Comment (fastrizwaan) [reply org.deluge_torrent.deluge.yaml:42]: Is it OK, if I don't want to use flatpak-builder-tools' pip generator?

Comment (barthalion) [reply org.deluge_torrent.deluge.yaml:42]: Yeah, whatever works for you. Just posted it so you know it exists.

File: org.deluge_torrent.deluge.yaml (hunk 12 above)
Comment (fastrizwaan) [reply org.deluge_torrent.deluge.yaml:16]: added cleanup for boost and libtorrent

File: org.deluge_torrent.deluge.yaml (hunk 4 above)
Comment (fastrizwaan) [reply org.deluge_torrent.deluge.yaml:38]: https://dev.deluge-torrent.org/wiki/Installing/Source, seeing the build instructions,  I could not find specific part of partial build of boost modules for deluge. 

```
If building libtorrent:
boost >= 1.34.1
```

File: org.deluge_torrent.deluge.yaml (hunk 11 above)
Comment (fastrizwaan) [reply org.deluge_torrent.deluge.yaml]: using rename-icon, copy-icon not required.

File: README.md (hunk 9 above)
Comment (fastrizwaan) [reply README.md]: Removed Readme.md as it is not required for flatpak building and packaging.

**Review by fastrizwaan — COMMENTED** [review]
Really appreciate your patience and guidance. Thank you!

**Comment by fastrizwaan** [comment]
bot, build org.deluge_torrent.deluge

**Comment by fastrizwaan** [comment]
Please review.

**Comment by barthalion** [comment]
Hi @cas--, would you be interested in (co-)maintaining Deluge on Flathub?

**Comment by cas--** [comment]
Yeah that'd be fine. Appreciate the work getting it here, thanks!

**Pull request closed without merging**